v3/src/components/inspector-panel.tsx

```diff
@@ -72,20 +72,27 @@ interface IInspectorPalette {
   children: ReactNode
   Icon?: ReactNode
   title?: string
-  paletteTop?: number
   panelRect?: DOMRect
   buttonRect?: DOMRect
   setShowPalette: (palette: string | undefined) => void
 }
 
-export const InspectorPalette = ({children, Icon, title, paletteTop = 0,  panelRect, buttonRect,
+export const InspectorPalette = ({children, Icon, title, panelRect, buttonRect,
      setShowPalette}:IInspectorPalette) => {
   const panelTop = panelRect?.top || 0
   const buttonTop = buttonRect?.top || 0
   const [paletteWidth, setPaletteWidth] = useState(0)
   const paletteRef = useRef<HTMLDivElement>(null)
+  const pointerRef = useRef<HTMLDivElement>(null)
   const viewportEl = document.getElementsByClassName("free-tile-row")?.[0]
   const [inBounds, setInBounds] = useState(isWithinBounds(paletteWidth, panelRect))
+  const paletteHeight = paletteRef.current?.offsetHeight
+  const tempPaletteTop = paletteRef.current?.getBoundingClientRect().top
+  const pointerTop = buttonTop - panelTop - 5
+  const pointerHeight = pointerRef.current?.offsetHeight
+  const pointerMidpoint = pointerHeight ? pointerTop + pointerHeight/2 : 13
+  const paletteTop = (tempPaletteTop && paletteHeight) &&
+    getPaletteTopPosition(tempPaletteTop, paletteHeight, pointerMidpoint)
 
   useEffect(()=> {
     const observer = new ResizeObserver(entries => {
```

Comment:
Two things here:
1. I think you need to deal with the possibility of `viewportEl` being `undefined` or `null`.
2. You should dispose of the observer in the return value of the effect.

Something like:
```
    const observer = viewportEl && new ResizeObserver(entries => {
      entries.forEach(entry => {
        if (panelRect) {
          setInBounds(isWithinBounds(paletteWidth, panelRect))
        }
      })
    })
    observer?.observe(viewportEl)
    return () => observer?.disconnect()
```